src/Database/LSMTree/Internal.hs

```diff
@@ -880,11 +879,11 @@ data CursorEnv m h = CursorEnv {
   , cursorReaders    :: !(Maybe (Readers.Readers m h))
     -- | The runs held open by the cursor. We must remove a reference when the
     -- cursor gets closed.
-  , cursorRuns       :: !(V.Vector (Run m h))
+  , cursorRuns       :: !(V.Vector (Ref (Run m h)))
```

Comment:
(not something we have to fix in this PR)

Reply:
K/ops files are not tracked separately to the runs. If a run reaches zero references, it will remove the k/ops file. So unless we do a larger re-design, we need to keep references to the runs, not the k/ops files.

I also don't think keeping the in-memory structures alive is a problem, since cursors should usually be quite short-lived.

Reply:
I think the right solution is for the 'RunReader' and 'Readers' to properly retain the 'BlobFile' of the 'Run' and 'WriteBufferBlobs' so that the 'BlobRefs' remain valid, until the 'RunReader' and 'Readers' are explicitly closed.

This would then simplify the 'Cursor' which would not need to keep seaprate references to the 'Run' or 'WriteBufferBlobs' at all, but would just need to close the 'Readers'.

Incidentally, 'Readers' and 'RunReader' have an inconsistent naming convention.

Reply:
I've added TODOs to this effect.